doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,382 @@
+- Last Updated: 2024-03-21
+- Discussion at:
+
+Abstract
+===========
+
+In SCION the endhosts have the option to choose inter-AS paths to forward packets to a destination.
+However some applications require more fine grained path selection like "Do not route traffic over devices
+produced by hardware manufacturer X or that run software Y with version < Z" for the intra-AS paths.
+This is useful for example if there exists a known bug in certain router versions that affect secure communication,
+or if an entity simply does not trust a certain manufacturer.
+This can also be seen as an enhancement for keeping traffic within a certain jurisdiction, e.g. by avoiding routers
+of a manufacturer from a certain country.
+
+Background
+===========
+
+`FABRID <https://netsec.ethz.ch/publications/papers/2023_usenix_fabrid.pdf>`_, is suggested as a solution to the
+aforementioned problem.
+An implementation of FABRID in SCIONLab allows for wider testing and evaluations of the protocol
+and gives other people the possibility to use it for their applications.
+FABRID also implicitly implements `EPIC <https://netsec.ethz.ch/publications/papers/Legner_Usenix2020_EPIC.pdf>`_,
+with a slightly little less accurate timestamp.
+
+Proposal
+========
+
+We introduce FABRID policies which can be thought of as additional path constraints that should be applied for intra-AS paths.
+The border routers use those policies to decide on the intra-AS path to forward, e.g. by using MPLS labels.
+Some FABRID policies are globally defined and others locally per AS.
+The AS network operator configures which global FABRID policies are supported for the local AS and can add additional local FABRID
+policies that are valid for this AS.
+A source endhost can then select a path together with FABRID policies and forward the FABRID packet over this path to a destination endhost.
+The destination endhost will then be able to recompute the path validator to verify that the packet had been forwarded over that inter-AS path.
+
+The implementation of FABRID allows for incremental deployment at router- and AS-level.
+This could lead to the situation that we cannot find an end-to-end FABRID-enabled path.
+In this case, the sending endhost can disable FABRID for that AS when sending a packet but will not have any of the guarantees provided by FABRID for that AS.
+
+Since each AS can create their own FABRID policies, the other ASes have to learn them.
+Those policies are only fetched on demand by the local control service and will be cached till end of their validity.
+This also makes sure that the beacon size does not grow too big.
+The source endhost does not have to do anything about that, he will learn the policies from its local control service
+which will either return the cached policies of query the remote control service.
+
+The proposal consists of a header design, namely two new Hop-by-Hop extension options, forwarding support in the routers,
```

Comment:
This design document specifies the details of the ...

Reply:
Mention here already that we plan to extend the design doc later on with ... e.g., path validation for the source end host etc.

Reply:
Also mention that we will expand this documentation (or maybe better: create separate design documents) also for MPLS support?